core/include/scipp/core/variable.h

```diff
@@ -216,6 +216,18 @@ template <class... Known> class VariableConceptHandle_impl {
   operator=(VariableConceptHandle_impl &&) = default;
   VariableConceptHandle_impl &
   operator=(const VariableConceptHandle_impl &other) {
+    if (*this && other) {
+      // Avoid allocation of new element_array if output is of correct shape.
+      // This yields a 5x speedup in assignment operations of variables.
+      auto &concept = **this;
+      auto &otherConcept = *other;
+      if (!concept.isView() && !otherConcept.isView() &&
```

Comment:
Looks good. Might be worth creating an explicit function to describe conditions here. `bool canCopy(const VariableConceptHandle&from, const VariableConceptHandle& to)`

Reply:
I could introduce a lambda but that does not improve much? I would avoid making this a free function since in this case I feel this would actually hurt readability --- `canCopy` does not say much and the reader would need to go look for the function --- since it would hide crucial information such that this is only done when not dealing with views.

I agree though that the mechanism is not nice and would like to find a better way.